### PR TITLE
#9 task.andrei.manya

### PR DESCRIPTION
**Pull request by AndreiManya**

### CODE REVIEW

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -65,7 +68,16 @@ function getPowerFunction(exponent) {
  *   getPolynom()      => null
  */
 function getPolynom() {
-    throw new Error('Not implemented');
+    let args = arguments;
+    if (args.length == 0 )
+        return null;
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js:73]: if block is on a new line use {}

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -104,10 +118,22 @@ function memoize(func) {
  * retryer() => 2
  */
 function retry(func, attempts) {
-    throw new Error('Not implemented');
+    return ()=>{
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js:123]: You don't need 2 exit condition for a for-loop. See image below

image

File: task/09-functions-n-closures-tasks.js [hunk 3]
```diff
@@ -65,7 +68,16 @@ function getPowerFunction(exponent) {
...
  */
 function getPolynom() {
-    throw new Error('Not implemented');
+    let args = arguments;
+    if (args.length == 0 )
+        return null;
+    return (x)=>{
+        let result = 0;
+        for (let i = 0; i < args.length; i++)
+            result += args[i] * Math.pow(x, args.length - i - 1);
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js:78]: if block is on a new line use {}

**Comment by AlexMokin** [comment]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Review by AlexMokin — COMMENTED** [review]
Please see the comments and fix accordingly. Also provide a travis ci link
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by AndreiManya** [comment]
https://travis-ci.org/github/AndreiManya/js-assignments/builds/710278612

**Comment by AndreiManya** [comment]
So, all this tasks fine?

**Comment by AlexMokin** [comment]
accepted

**Pull request closed without merging**